File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = 0;
        for(int num : nums){
            max = Math.max(max,num);
        }
        int cur = 0, ans = 0;
        for(int num : nums){
            if(num == max){
                cur++;
                ans = Math.max(ans,cur);
            }else{
                cur = 0;
            }
        }
        return ans;
    }
}

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        a = max(nums)
        cnt = 0 
        ans = 0
        for num in nums:
            if num == a:
                cnt += 1
                ans = max(ans,cnt)
            else:
                cnt = 0
        
        return ans